File: selection_sort.js
function selectionSort(array){
	var min;
	var temp;
	for (var i = 0; i < array.length; i++){
		min = i;
		for(var j = i + 1; j < array.length; j++){
			if(array[j] < array[min]){
				min = j;
			}
		}
		if(i != min){
			temp = array[i];
			array[i] = array[min];
			array[min] = temp;
		}
	}
	return array;
}


var myArray = [5,4,3,2,1];
selectionSort(myArray);
console.log(myArray);

File: selection_sort.rb
def selection_sort(arr)
	for i in 0...arr.length-1
		min = i
		j = i
		for j in j...arr.length
			min = j if arr[j] < arr[min]		
		end

		unless i == min
			arr[i], arr[min] = arr[min], arr[i]
		end

	end
	arr
end

arr = Array.new(10) { rand(1..99) }

puts "before #{arr}"
selection_sort(arr)
puts "after #{arr}"